### ISSUE TITLE
PFFile and PFLogger can cause undefined execution

### ISSUE THREAD

**Issue by MatrixSenpai** [parse-community/Parse-SDK-iOS-OSX#1325]
PFFile and PFLogger should probably be renamed. They both conflict with classes from `PhotoFoundation` in iOS

```
objc[6468]: Class PFFile is implemented in both /System/Library/PrivateFrameworks/PhotoFoundation.framework/PhotoFoundation and /private/var/containers/Bundle/Application/2CDAE808-924C-48CC-8937-98C5A1D22A56/<redacted>.app/Frameworks/Parse.framework/Parse. One of the two will be used. Which one is undefined.
objc[6468]: Class PFLogger is implemented in both /System/Library/PrivateFrameworks/PhotoFoundation.framework/PhotoFoundation and /private/var/containers/Bundle/Application/2CDAE808-924C-48CC-8937-98C5A1D22A56/<redacted>.app/Frameworks/Parse.framework/Parse. One of the two will be used. Which one is undefined.
```

**Comment by flovilmart** [comment]
Ah yeah? That’s really annoying. Not sure what to rename them though :/

**Comment by MatrixSenpai** [comment]
What about PFFileObject and PFSystemLogger?

**Comment by flovilmart** [comment]
@MatrixSenpai this makes sense, I'm kinda p155ed for `PFFile` as this will require some changes in all code bases.

**Comment by flovilmart** [comment]
what's the result of `npm ls parse-server`?

**Comment by RyanBertrand** [comment]
This seems to be crashing our app on iOS 12 which is releasing in the next month or so.  What is the game plan to rename these?  Let me know if I can help.

Thanks

**Comment by flovilmart** [comment]
Well, you can always open a Pull Request with the fix.

**Comment by pollini** [comment]
Any news on this issue? I've been running into this too while testing and preparing for iOS 12. Since my app isn't using the PhotoFoundation framework, I haven't seen any impacts though so far...

**Comment by jtressle** [comment]
@flovilmart what are the steps to fixing this?

If I replace all PFFile with PFFileObject on a fork of the parse git, will pointing pods to my git work?  Or do I need to rebuild something?

Thanks,

**Comment by RyanBertrand** [comment]
@jtressle  I did it for one of our apps.  https://github.com/RyanBertrand/Parse-SDK-iOS-OSX

Podfile:
```pod 'Parse', :git => 'https://github.com/RyanBertrand/Parse-SDK-iOS-OSX', :branch => 'rename-PFFile-and-PFLogger'```

I don't have plans to actively maintain it on the fork.  If it doesn't take much time, we can do a PR.

Thanks,
Ryan

**Comment by jtressle** [comment]
@RyanBertrand awesome thanks!  I can confirm your changes work.  

Unfortunately, I'm rewriting my app quick from Obj-C to Swift so I can't check the ability to pull a file from a database.  This change shouldn't affect any connectivity to the database right?

**Comment by flovilmart** [comment]
Please, do a PR, that will most probably help everyone

**Comment by AlexGingell** [comment]
I'm having difficulties compiling your branch according to the instructions @RyanBertrand - is there any way that you (or anyone else) could share compiled Parse.framework and Bolts.framework files including all iOS architectures?  I don't need any other targets and these seem to be the ones hamstringing the process (tvOS etc). Many thanks for any help!

**Comment by fxchou123** [comment]
Same issue here.

**Comment by mtrezza** [comment]
> What about PFFileObject and PFSystemLogger?

Apple actually has a guideline for [Preventing Name Conflicts](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/LoadingCode/Tasks/NameConflicts.html). According to that the names should be:

`<BundleIdentifierCamelCased><Class>`, e.g. `ComParseIosFile` instead of `PFFile`

And to avoid having to use the full, prefixed symbol names in source code:
`#define ParseFile ComParseIosFile`

**Comment by MatrixSenpai** [comment]
@mtrezza you're right about the naming guide, I was just throwing ideas out. However, even using `#define` could cause unexpected behavior, unless those are made available just before compile time. I'm not familiar with how all that works

**Comment by JEKos** [comment]
Can someone make a pull request with the fix please?

**Comment by ZacharyKhan** [comment]
Also experiencing this error in the log, but it isn't fatal so I can look past it for now. I like the approach @mtrezza suggested. Renaming `PFFile` to `PFile` or `ParseFile` would be best in my opinion. 

If I get some spare time I'll fork and submit a PR this week.

**Comment by Woildan** [comment]
@ZacharyKhan I've submitted a PR with this already #1355.
CI fails but I don't think it is related to the changes. None of the other PR awaiting passes CI.

I don't know what more to do now.

**Comment by ccapener** [comment]
After the @Woildan commit was merged, I cloned the master to my local machine, built the SDK, and updated my podfile to point to the local version.  Then ran `pod update Parse`.  The new files were brought into my Xcode project.  Building the project resulted in the expected errors of `PFFile` not being defined.  I then changed all occurrences of `PFFile` in my code to `PFFileObject`.  All went well except when creating a new `PFFileObject`:

`let imageFile = PFFileObject(name: "postImage.jpg", data: imageData)`

This resulted in an error: _Argument passed to call that takes no arguments_

Comparing the `PFFileObject` .h and .m files with the old `PFFile` files did not turn up any obvious changes the would result in this error.  Ideas?

**Comment by Woildan** [comment]
@ccapener This is due to the automatic generation of Swift API from Obj-C. We should have renamed the factory method to reflect the new class names so proper constructor are generated for Swift.

Right now you can change your code to

`let imageFile = PFFileObject.file(withName: "postImage.jpg", data: imageData)`

I'll try to find some time to submit a new PR that solves this + update the changelog / migration guide.

**Comment by ccapener** [comment]
@Woildan Thanks for the quick response. It fixed the problem.  And many thanks for submitting the code for `PFFileObject()`!

**Comment by wdcurry** [comment]
Which pod should we be currently pointing to in order to grab the corrected bits?

**Comment by MatrixSenpai** [comment]
@wdcurry #1355 merged into master. You can pull the latest version

**Comment by wdcurry** [comment]
> @wdcurry #1355 merged into master. You can pull the latest version

Thank you, that did the trick, much obliged!
